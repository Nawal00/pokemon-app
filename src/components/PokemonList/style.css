.pokemon-list {
  width: 300px;
  background-color: #7fffd4;
  border-radius: 5px;
  padding: 0 0.5rem;
  cursor: pointer;
  text-align: center;
}
