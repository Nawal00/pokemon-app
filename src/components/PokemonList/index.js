import PropTypes from "prop-types";
import "./style.css";

const PokemonList = ({ name, fetchPokemonInfo }) => {
  return (
    <div
      className="pokemon-list"
      role="list"
      key={name}
      onClick={() => fetchPokemonInfo(name)}
    >
      <h3>{name}</h3>
    </div>
  );
};

export default PokemonList;

PokemonList.propTypes = {
  name: PropTypes.string.isRequired,
  fetchPokemonInfo: PropTypes.func.isRequired,
};
