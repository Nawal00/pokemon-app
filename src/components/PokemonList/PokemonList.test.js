import { render, screen, fireEvent } from "@testing-library/react";
import PokemonList from "./";

const props = {
  name: "pikachu",
  fetchPokemonInfo: jest.fn(),
};

describe("Given the `PokemonList` component", () => {
  it("should render with correct props", () => {
    const { container } = render(<PokemonList {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe("When user clicks on a pokemon name", () => {
    it("should call `fetchPokemonInfo` with correct params", () => {
      render(<PokemonList {...props} />);

      const pokemon = screen.getByRole("list");

      fireEvent.click(pokemon);

      expect(props.fetchPokemonInfo).toHaveBeenCalledWith("pikachu");
    });
  });
});
