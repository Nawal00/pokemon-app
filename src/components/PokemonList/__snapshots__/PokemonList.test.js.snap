// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the \`PokemonList\` component should render successfully 1`] = `
<div>
  <div
    class="pokemon-list"
    role="list"
  >
    <h3>
      pikachu
    </h3>
  </div>
</div>
`;
